tests/Infrastructure.Tests/Contracts/RecentContractRepositoryTests.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Linq;
+using Application.Contracts;
+using Domain.Contracts;
+using Domain.Users;
+using Infrastructure.Databases;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Tests.Contracts;
+
+[Collection("DatabaseTests")]
+public class RecentContractRepositoryTests : IClassFixture<TestDatabaseFixture>, IDisposable
+{
+    private readonly EFRecentContractRepository _cut;
+    private readonly EFDatabaseContext _context;
+
+    public RecentContractRepositoryTests(TestDatabaseFixture fixture)
+    {
+        _context = fixture.CreateContext();
+        _cut = new EFRecentContractRepository(_context, Mock.Of<ILogger<EFRecentContractRepository>>());
+        _context.Database.BeginTransaction();
+    }
+
+    [Fact]
+    public void AddRecent_AddsTheContractCorrectly()
+    {
+        // Arrange
+        var user1 = new User() { Name = "User1" };
+        AddUserToContext(user1);
+
+        var contract1 = new Contract() { Name = "1" };
+        _context.Contracts.Add(contract1);
+
+        // Act
+        _cut.AddRecent(user1.Name, contract1);
+        IList<RecentlyViewedContract> contracts = _cut.FetchRecentContracts(user1.Name);
+
+        // Assert
+        contracts.First().ContractId.Should().Be(contract1.Id);
+    }
+
+    [Fact]
+    public void AddRecent_ReAddingAnExistingContractUpdatesTheTimeCorrectly()
+    {
+        // Arrange
+        var user1 = new User() { Name = "User1" };
+        AddUserToContext(user1);
+
+        var contract1 = new Contract();
+        AddContractToContext(contract1);
+
+        // Act
+        _cut.AddRecent(user1.Name, contract1);
+        var timeBeforeReAdd = _context.Users.First(user => user.Name == user1.Name).RecentlyViewContracts
+            .First(recentContract => recentContract.ContractId == contract1.Id)
+            .LastViewed;
+        _cut.AddRecent(user1.Name, contract1);
+        var timeAfterReAdd = _context.Users.First(user => user.Name == user1.Name).RecentlyViewContracts
+            .First(recentContract => recentContract.ContractId == contract1.Id)
+            .LastViewed;
+
+        // Assert
+        timeBeforeReAdd.Should().BeBefore(timeAfterReAdd);
+    }
+
+    [Fact]
+    public void RemoveRecent_RemovesCorrectContract()
```

Comment:
nit:
Same here as the first comment in this file, you shouldn't just say "Does the correct thing", you should specify what correct means:
```suggestion
    public void RemoveRecent_RemovesSpecifiedContract_WhenThereAreMultipleRecents()
```